sdk/communication/communication-common/test/public/node/entraCommunicationTokenCredential.spec.ts

```diff
@@ -58,6 +63,136 @@ describe("Entra CommunicationTokenCredential", function () {
     getTokenSpy.restore();
   });
 
+  it("Token exchange is called with default scopes when not passing scopes on AzureCommunicationTokenCredential", async function () {
+    const scope = successApiMock();
+    const defaultScopes = ["https://communication.azure.com/clients/.default"];
+    const getTokenSpy: sinon.SinonSpy = sinon.spy(tokenCredential, "getToken");
+
+    const entraTokenCredentialOptions: EntraCommunicationTokenCredentialOptions = {
+      resourceEndpoint: resourceEndpoint,
+      tokenCredential: tokenCredential,
+    };
+
+    const credential = new AzureCommunicationTokenCredential(entraTokenCredentialOptions);
+
+    const tokenResult = (await credential.getToken()).token;
+    assert.strictEqual(tokenResult, acsToken);
+    assert.isTrue(getTokenSpy.alwaysCalledWithExactly(defaultScopes, undefined));
+    assert.isTrue(getTokenSpy.callCount > 0);
+    assert.isTrue(scope.isDone());
+    getTokenSpy.restore();
+  });
+
+  it("Token exchange is not called again when Entra token stays the same", async function () {
+    let scope = successApiMock();
+    const entraTokenCredentialOptions: EntraCommunicationTokenCredentialOptions = {
+      resourceEndpoint: resourceEndpoint,
+      tokenCredential: tokenCredential,
+      scopes,
+    };
+    const entraTokenCredential = new EntraTokenCredential(entraTokenCredentialOptions);
+    const exchangeTokenSpy = sinon.spy(entraTokenCredential as any, "exchangeEntraToken");
```

Comment:
Why do we need a spy on the internal function when we're already using the api mock?